components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/WorkflowUtils.java

```diff
@@ -165,4 +180,132 @@ public static void sendNotificationAfterWFComplete(WorkflowDTO workflowDTO, Stri
+        apiStateWorkFlowDTO.setMetadata("ApiName", apiStateWorkFlowDTO.getApiName());
+        apiStateWorkFlowDTO.setMetadata("ApiVersion", apiStateWorkFlowDTO.getApiVersion());
+        apiStateWorkFlowDTO.setMetadata("ApiProvider", apiStateWorkFlowDTO.getApiProvider());
+        apiStateWorkFlowDTO.setMetadata("Invoker", apiStateWorkFlowDTO.getInvoker());
+        apiStateWorkFlowDTO.setMetadata("TenantId", String.valueOf(apiStateWorkFlowDTO.getTenantId()));
+
+        apiStateWorkFlowDTO.setProperties("action", apiStateWorkFlowDTO.getApiLCAction());
+        apiStateWorkFlowDTO.setProperties("apiName", apiStateWorkFlowDTO.getApiName());
+        apiStateWorkFlowDTO.setProperties("apiVersion", apiStateWorkFlowDTO.getApiVersion());
+        apiStateWorkFlowDTO.setProperties("apiProvider", apiStateWorkFlowDTO.getApiProvider());
+        apiStateWorkFlowDTO.setProperties("currentState", apiStateWorkFlowDTO.getApiCurrentState());
+    }
+
+    /**
+     * Complete the lifecycle state change workflow
+     *
+     * @param workflowDTO Workflow DTO object
+     * @throws WorkflowException Exception when completing the workflow
+     */
+    protected static void completeStateChangeWorkflow(WorkflowDTO workflowDTO) throws WorkflowException {
+
+        String externalWorkflowRef = workflowDTO.getExternalWorkflowReference();
+        try {
+            ApiMgtDAO apiMgtDAO = ApiMgtDAO.getInstance();
+            Workflow workflow = apiMgtDAO.getworkflowReferenceByExternalWorkflowReference(externalWorkflowRef);
+            String apiName = workflow.getMetadata("ApiName");
+            String action = workflow.getMetadata("Action");
+            String providerName = workflow.getMetadata("ApiProvider");
+            String version = workflow.getMetadata("ApiVersion");
+            String invoker = workflow.getMetadata("Invoker");
+            String currentStatus = workflow.getMetadata("CurrentState");
+            int tenantId = workflowDTO.getTenantId();
+            try {
+                //tenant flow is already started from the rest api service impl. no need to start from here
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(invoker);
+                Registry registry =
+                        ServiceReferenceHolder.getInstance().getRegistryService().getGovernanceUserRegistry(invoker,
+                                tenantId);
+                APIIdentifier apiIdentifier = new APIIdentifier(providerName, apiName, version);
+                GenericArtifact apiArtifact = APIUtil.getAPIArtifact(apiIdentifier, registry);
+                if (WorkflowStatus.APPROVED.equals(workflowDTO.getStatus())) {
+                    String targetStatus;
+                    apiArtifact.invokeAction(action, APIConstants.API_LIFE_CYCLE);
+                    targetStatus = apiArtifact.getLifecycleState();
+                    if (!currentStatus.equals(targetStatus)) {
```

Comment:
Can the currentStatus become null?

Reply:
Current API state is set to the workflow DTO [here](https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java#L4904). hence, the currentStatus cannot be null..